### ISSUE TITLE
Issues running on Windows DMD32 D Compiler v2.074.1

### ISSUE THREAD

**Issue by Poyeyo** [capnproto/capnproto-dlang#3]
Using it as a library:

    $ dub
    Fetching capnproto-dlang 0.1.2 (getting selected version)...
    Fetching openssl 1.1.5+1.0.1g (getting selected version)...
    Fetching collie 0.9.7 (getting selected version)...
    Performing "debug" build using dmd for x86.
    capnproto-dlang 0.1.2: building configuration "lib"...
    ..\..\..\Users\Poyeyo\AppData\Roaming\dub\packages\capnproto-dlang-0.1.2\capnproto-dlang\source\capnproto\FileDescriptor.d(38,34): Error: cannot implicitly convert expression (this.file.size()) of type ulong to uint
    dmd failed with exit code 1.

Running the test suite:

    $ dub test
    Generating test runner configuration 'capnproto-dlang-test-lib' for 'lib' (library).
    Excluding package.d file from test due to https://issues.dlang.org/show_bug.cgi?id=11847
    Performing "unittest" build using dmd for x86.
    capnproto-dlang 0.1.2+commit.15.g3f02106: building configuration "capnproto-dlang-test-lib"...
    Linking...
    Running .\capnproto-dlang-test-lib.exe
    
    core.exception.OutOfMemoryError@src\core\exception.d(696): Memory allocation failed
    ----------------
    Program exited with code 1

**Comment by ThomasBrixLarsen** [comment]
I think that at least the first is already fixed. Could you try using "~master" in dub.selections.json instead of "0.1.2"?

**Comment by Poyeyo** [comment]
Confirmed, the first one is fixed in master branch.

**Comment by ThomasBrixLarsen** [comment]
Excellent.

I think the second issue might be memory addressing related as dub test only allocates 7,16 MiB on my system. Can you try to build for 64 bit? Something like dub test --arch=x86_64 I imagine.

**Comment by Poyeyo** [comment]
Yes, it works in 64 bit mode. Is that an issue in the lib or in d runtime?

**Comment by ThomasBrixLarsen** [comment]
I've been meaning to look into this. I'm not sure where the issues is at this point. I will have to find a Windows machine to test on somewhere.
